### PR TITLE
[PF-2145] Null enhancements

### PR DESCRIPTION
**Pull request by iliadsh**
Using java's null as the value in the response resulted in spring thinking the value was unset, so replacing that with an explicit json null.

Also, nulls are stripped when setting in the db to act more like a real delete operation when setting a property to null.

### CODE REVIEW

**Review by cbookg — APPROVED** [review]

File: service/src/main/java/bio/terra/user/db/ProfileDao.java [hunk 1]
```diff
@@ -85,7 +86,7 @@ private String getPropertySingle(String userId, List<String> path) {
             return rs.getString("value");
           });
     } catch (EmptyResultDataAccessException e) {
-      return "{}";
+      return CollectionUtils.isEmpty(path) ? "{}" : null;
```
Comment (yuhuyoyo) [inline service/src/main/java/bio/terra/user/db/ProfileDao.java:91]: could you explain why returning {} when path is empty?

also annotate the method and getProperty with @nullable now you are retuning null in some cases.

**Review by yuhuyoyo — APPROVED** [review]

File: service/src/main/java/bio/terra/user/db/ProfileDao.java (hunk 1 above)
Comment (iliadsh) [reply service/src/main/java/bio/terra/user/db/ProfileDao.java:91]: This combined with not being able to directly set the root means that clients can expect the profile object to always exist.

**Comment by yuhuyoyo** [comment]
btw are you or Chris admin of this repo? can we set up branch protection?

**Comment by iliadsh** [comment]
LOL we just did. Chris is admin I believe.

File: service/src/main/java/bio/terra/user/db/ProfileDao.java (hunk 1 above)
Comment (yuhuyoyo) [reply service/src/main/java/bio/terra/user/db/ProfileDao.java:91]: oh i see. given you create a row when non exist, does it still throw emptyReuslt exception when path is empty? I thought it would just return {}

File: service/src/test/java/bio/terra/user/db/ProfileDaoTest.java [hunk 2]
```diff
@@ -49,7 +49,7 @@ void clobberSet() throws Exception {
     assertEquals(workspaces, profileDao.getProperty(userId, path));
 
     profileDao.setProperty(userId, path, "null");
-    assertEquals("null", profileDao.getProperty(userId, path));
+    assertEquals(null, profileDao.getProperty(userId, path));
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/user/db/ProfileDaoTest.java:60]: add test for empty path with the new change.

File: service/src/main/java/bio/terra/user/db/ProfileDao.java (hunk 1 above)
Comment (iliadsh) [reply service/src/main/java/bio/terra/user/db/ProfileDao.java:91]: Atm the row create is only attempted on set so this exception is possible when trying to read for the first time. 

Do you think it should also be created on get? I did it like this to save on queries, but one basically no-op query shouldn't really matter so...

Comment (yuhuyoyo) [reply service/src/main/java/bio/terra/user/db/ProfileDao.java:91]: i see. thanks for explaining. please add a comment
